client/src/plugins/camunda-plugin/deployment-tool/__tests__/DeploymentConfigOverlaySpec.js

```diff
@@ -106,25 +107,20 @@ describe('<DeploymentConfigOverlay>', () => {
         validator
       }, mount);
 
-      // when
-      setTimeout(() => {
-
-        // delayed execution because it is async that the deployment
-        // tool knows if the authentication is necessary
-        instance.isOnBeforeSubmit = true;
-        wrapper.find('.btn-primary').simulate('submit');
-      });
+      // delayed execution because it is async that the deployment
+      // tool knows if the authentication is necessary
+      instance.isOnBeforeSubmit = true;
+      wrapper.find('.btn-primary').simulate('submit');
 
       // then
-      setTimeout(() => {
+      waitFor(() => {
```

Comment:
We should `await` this call as otherwise the tests reports green even if the callback fails.

Reply:
```suggestion
      await waitFor(() => {
```